libopenage/util/fixed_point.h

```diff
@@ -589,6 +589,19 @@ constexpr double hypot(openage::util::FixedPoint<I, F> x, openage::util::FixedPo
 	return x.hypot(y);
 }
 
+template <typename I, unsigned F>
+constexpr openage::util::FixedPoint<I, F> pow(openage::util::FixedPoint<I, F> base, double exponent) {
+	// Handle special cases: 0^0, 1^x, x^0
+	if (base == 0.0 && exponent == 0.0) {
+		return openage::util::FixedPoint<I, F>::quiet_NaN(); // Undefined; return NaN
+	} else if (base == 1.0 || exponent == 0.0) {
```

Comment:
why compare to doubles in the if conditions when we could compre to fixed point values instead?